pjrt/third_party/re2/re2.cc

```diff
@@ -0,0 +1,28 @@
+#include "absl/strings/string_view.h"
+#include "absl/strings/numbers.h"
+
+#include "re2/re2.h"
+
+extern "C" {
+  bool isTfOpName(const char *op_name, size_t len) {
+    absl::string_view op_name_view(op_name, len);
+    static const LazyRE2 kTfOpNameRegEx = {"[A-Za-z0-9.][A-Za-z0-9_.\\/>-]*"};
+    return RE2::FullMatch(op_name_view, *kTfOpNameRegEx);
+  }
+
+  bool isJaxOpType(const char *op_type, size_t len) {
+    absl::string_view op_type_view(op_type, len);
```

Comment:
not a big fan of special casing. Also this is used to strip information ouptutted by the compiler, so I think we don't want to do that.  

overall I think we would be better by not trying to guess the source framework based on the casing of the operator name, and drop this dependency on re2.

Reply:
Looking into this bit; this is based directly on how XLA parses: https://github.com/openxla/xla/blob/cb5bdb601bdb352afaa804b5b4a8826734772e3b/xla/tsl/profiler/utils/tf_op_utils.cc#L129

I agree it would be best to get rid of the `re2` and `abseil-cpp` dependencies.

Reply:
I understand XLA is doing weird stuff, but I don't think we should reproduce that part. The goal is to parse the files generated while profiling ZML, not tensorflow or jax. Also I believe this way of handling the input is very brittle, because it can categorize ZML operators as tensorflow / jax and apply logic where it's not warranted.
I'd rather remove this logic altogether.

Reply:
I don't think this was addressed

Reply:
Gotcha, I'll just remove this logic entirely then.

Reply:
Sorry, somehow I missed the comment you'd left on this 4 days ago after I'd initially revised to remove the dependencies on `re2`/`abseil-cpp`.